rust/headless-client/src/lib.rs

```diff
@@ -267,12 +276,14 @@ pub fn run_only_headless_client() -> Result<()> {
     );
     // TODO: this should be added dynamically
     session.set_dns(platform::system_resolvers().unwrap_or_default());
+    platform::notify_service_controller()?;
 
     let result = rt.block_on(async {
+        let mut interface = interface::InterfaceManager::default();
         let mut signals = Signals::new()?;
 
         loop {
-            match future::select(pin!(signals.recv()), pin!(on_disconnect_rx.recv())).await {
+            match future::select(pin!(signals.recv()), pin!(cb_rx.recv())).await {
```

Comment:
is `pin` needed here? aren't these `Unpin`?

Reply:
🤷‍♀️ If I try to remove either of those `pin!()` macros it won't compile.
I don't know how pinning works so I could be overlooking something.